test/e2e/autoscaling_utils.go

```diff
@@ -236,17 +236,18 @@ func (rc *ResourceConsumer) GetReplicas() int {
 	return 0
 }
 
-func (rc *ResourceConsumer) WaitForReplicas(desiredReplicas int) {
+func (rc *ResourceConsumer) WaitForReplicas(desiredReplicas int) bool {
```

Comment:
I think the more go-stylistic way would be to return an `err` on error. `nil` on success.